### ISSUE TITLE
Does SourceLink not work for embedded pdb files?

### ISSUE THREAD

**Issue by christcottrell** [dotnet/sourcelink#798]
Before I hear back that they should be in snupkg packages, I need to hear back on whether they are supported with the pdb files embedded.  This is one of our simplest use cases we have here at Deloitte and it's not working. 

The code is being pushed to a Github within the Deloitte organization so the source code and the symbols are not public (hence not being able to use existing symbol servers that are public.  From the nuget package explorer, everything looks ok.

![image](https://user-images.githubusercontent.com/53835198/153434668-d8023703-1c09-4150-9525-1424b48021d1.png)

The only thing I can think of is that the authentication to the Github repo is not in place. What I do have in my solution is a nuget.config that registers the nuget source and I am able to pull from our nuget repo from package manager without issues. But does the pulling of the source code use those nuget.config source settings?  Has this been tested against non public code repos? 

Thanks!

**Comment by christcottrell** [comment]
This is from 2 years ago? Really? And there is still no proper documentation on it. So when I'm rating my experience with Microsoft to be very poor, THIS is why. You build bad or incomplete products and don't document anything. Most have wasted time building and setting up something that just doesn't work or a simple set of documents explaining the best way would help.  I can go on and on with examples, but THIS is why you are laughed at as a trusted organization.  You also do too much leading the things you do to be not all put together. 

https://github.com/MicrosoftDocs/azure-devops-docs/issues/3709

**Comment by christcottrell** [comment]
This is likely another question and I don't want to distract from the original question, but if a package is consuming another package that is also tied to Source Link. So we have package A being used by a consumer. Package A uses package B. When debugging from the consumer and we're able to get into package A, does it allow that debugging to flow into package B? Thanks!

**Comment by tmat** [comment]
> does it allow that debugging to flow into package B? T

Yes, it should work as long as each of the packages is built with Source Link. Each assembly (.dll/.exe) has its own Source Link record in its PDB and is handled separately. Their sources can be in different repos of different git hosts.

**Comment by chuckries** [comment]
@tmat thanks, I assumed Embedded PDB.

@christcottrell I think the issue then is that the debugger is not finding the PDB's from within the nuget package. Try this:

1. Open Modules Window via Debug > Windows > Modules.
2. Right click on CMP.Validation.Service > Load Symbols
3. You should see a File Open Dialog
4. Browse to PDB in the NuGet Package within the NuGet Package Cache (defualt location is %USERPROFILE%\.nuget\pacakges

Ensure that the PDB is loaded via the modules window. If the PDB is not loaded, Source Link cannot work.

**Comment by mattnewport** [comment]
Thanks @chuckries for the solution here, I ran into the same problem and manually locating the .pdb file worked.

Now I'm wondering if there's a way to make this work automatically? Is there a good way to make this seamless for private packages?

**Comment by mattnewport** [comment]
Switching the project settings so that the symbols are embedded into the .dll rather than placed in a separate .pdb seems to solve the problem and is an acceptable tradeoff for our use case so I think that's what we'll do for internal packages going forward.

**Comment by emrecoskun705** [comment]
Hi @mattnewport, How can I embed my pdb into dlls?
